packages/teuchos/numerics/src/Teuchos_SerialDenseMatrix.hpp

```diff
@@ -398,55 +392,67 @@ class SerialDenseMatrix : public CompObject, public BLAS<OrdinalType, ScalarType
     ScalarType alpha = ScalarTraits<ScalarType>::zero() );
   void deleteArrays();
   void checkIndex( OrdinalType rowIndex, OrdinalType colIndex = 0 ) const;
-  OrdinalType numRows_;
-  OrdinalType numCols_;
-  OrdinalType stride_;
-  bool valuesCopied_;
-  ScalarType* values_;
 
+  static ScalarType*
+  allocateValues(const OrdinalType numRows,
+                 const OrdinalType numCols)
+  {
+    const size_t size = size_t(numRows) * size_t(numCols);
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wvla"
+    return new ScalarType[size];
+#pragma GCC diagnostic pop
+  }
+
+  OrdinalType numRows_ = 0;
+  OrdinalType numCols_ = 0;
+  OrdinalType stride_ = 0;
+  bool valuesCopied_ = false;
+  ScalarType* values_ = nullptr;
```

Comment:
Right!  Nice!  C++11 added in-class non-static member initialization so now you can use default constructors a lot more..  I love C++11 :-)